File: 03_recursion/c++/03_factorial.cpp
#include <iostream>

using namespace std;

int fact(int num){
	if(num == 1){
		return num;
	}
	else{
		return num * fact(num-1);
	}
}

int main(int argc, char *argv[]) {
	cout << fact(5) << endl; 	
}

File: 03_recursion/java/03_factorial.java
import java.util.*;
import java.lang.*;
import java.io.*;

class grokkingAlgorithms
{	
	public static int fact(int num){
		if(num == 1)
			return num;
		else
			return num * fact(num-1);
	}

	public static void main (String[] args) throws java.lang.Exception
	{
		System.out.println(fact(5));
	}
}